experimental/dds/tree2/src/feature-libraries/schemaBuilderBase.ts

```diff
@@ -16,9 +17,44 @@ import {
 	TypedSchemaCollection,
 	FlexList,
 	Unenforced,
+	Any,
 } from "./typed-schema";
 import { FieldKind } from "./modular-schema";
 
+/**
+ * Configuration for a SchemaBuilder.
+ * @alpha
+ */
+export interface SchemaBuilderOptions<TScope extends string = string> {
+	/**
+	 * Prefix appended to the identifiers of all {@link TreeSchema} produced by this builder.
+	 * Use of [Reverse domain name notation](https://en.wikipedia.org/wiki/Reverse_domain_name_notation) or a UUIDv4 is recommended to avoid collisions.
+	 */
+	scope: TScope;
+
+	/**
+	 * Name used to refer to this builder in error messages. Has no impact on the actual generated schema. Defaults to scope.
+	 */
+	name?: string;
+
+	/**
+	 * Optional configuration for "linting". See {@link SchemaLintConfiguration}. Currently defaults to enabling all lints.
+	 */
+	lint?: Partial<SchemaLintConfiguration>;
+
+	/**
+	 * Libraries to include in this one.
```

Comment:
Is it worth noting that the leaf domain is always included?

Reply:
It is not included in this base class. Doing so would be invalid (a cycle) as the leaf domain is implemented using this class.